llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -1121,6 +1121,40 @@ static void computeKnownBitsFromOperator(const Operator *I,
       break;
     }
 
+    Value *V;
+    // Handle bitcast from floating point to integer.
+    if (match(const_cast<Operator *>(I), m_ElementWiseBitCast(m_Value(V))) &&
+        V->getType()->isFPOrFPVectorTy()) {
+      KnownFPClass Result = computeKnownFPClass(V, fcAllFlags, Depth + 1, Q);
+      if (Result.SignBit) {
+        if (*Result.SignBit)
+          Known.makeNegative();
+        else
+          Known.makeNonNegative();
+      }
+
+      Type *FPType = V->getType()->getScalarType();
+      if (FPType->isIEEELikeFPTy()) {
+        int MantissaWidth = FPType->getFPMantissaWidth();
+        if (MantissaWidth != -1) {
+          if (Result.isKnownOnly(fcInf)) {
```

Comment:
Can do the same, plus the sign bit for fcPosInf/fcNegInf

Reply:
Isn't it handled by the code above?
```
      if (Result.SignBit) {
         if (*Result.SignBit)
           Known.makeNegative();
         else
           Known.makeNonNegative();
       }
```

Reply:
Isn't it handled by the code above?
```
      if (Result.SignBit) {
         if (*Result.SignBit)
           Known.makeNegative();
         else
           Known.makeNonNegative();
       }
```